packages/python/plotly/plotly/express/_core.py

```diff
@@ -869,6 +885,10 @@ def _get_reserved_col_names(args, attrables, array_attrables):
     return reserved_names
 
 
+def _isinstance_listlike(x):
+    return isinstance(x, list) or isinstance(x, np.ndarray) or isinstance(x, pd.Series)
```

Comment:
a slightly more permissive approach would be something like `False` if it's a string, then `True` if `iter(x)` doesn't raise an exception. This would allow through more stuff like `Series` from non-Pandas libraries etc. Of course it would allow through stuff like `dict`s but those would fail later on anyway.

Reply:
Ah but we do a `tuple` check later on. Hmm maybe we can check for a length-2-tuple first and then do this ?

Reply:
I was trying to understand which objects can be added in a column of a dataframe, and it occurred to me that the only reason why we need `px.Constant` is that we error out on
```
px.scatter(x=[1, 2, 3], y=1])
```
or 
```
px.bar(x='Monday', y=[1, 2, 3])
```
because if we did not catch this we could just the let the value be added for all rows. No regrets here @nicolaskruchten ? I think I'm fine with raising and having px.Constant since this is more robust to typos, but it could also be a lighter way of calling functions so I'd like to be sure we discuss this.

Reply:
well `y=1` is used to refer to column names that are `1` already...

Reply:
I suppose we could be "smart" about it and automatically wrap `y=1` in a `px.Constant()` if there is no data frame or if there is no df column named `1` ... ?

Reply:
if we start going down that road we might be able to make `identity` be automatic too :)

Reply:
well these are suggestions, but personally I'm happier if px is not too magical, ie if it's not the viz equivalent of pandas (where almost everything is possible but you don't really understand why).

Reply:
regarding your initial suggestion I implemented the more permissive version of the function.

Reply:
Cool, this works for me. I also worry about too much magic :)